### PR TITLE
Fixes #12702 AUTOMAGDROP weapons no longer have a loud buzzer as their go to mag drop sound.

### LINKED ISSUE #12702: Every weapon with AUTOMAGDROP plays smg_empty_alarm.ogg even if it's nonsensical for it to play a loud fucking buzzer
#### Description of issue

```

/obj/item/weapon/gun/projectile/afterattack(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, flag, struggle = 0)
    ..()
    if(!chambered && stored_magazine && !stored_magazine.ammo_count() && gun_flags &AUTOMAGDROP) //auto_mag_drop decides whether or not the mag is dropped once it empties
        RemoveMag(user)
        playsound(user, 'sound/weapons/smg_empty_alarm.ogg', 40, 1)
    return

```

this is not the place to do ``` playsound(user, 'sound/weapons/smg_empty_alarm.ogg', 40, 1 ```

#### Difference between expected and actual behavior

my deagle brand deagle not to sound like an alarm clock when it rounds outta dakka

it indeed did sound like an alarm clock

#### Steps to reproduce 

get gun with automagdrop shoot gun it yaps at you 

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [ ] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [ ] Issue happened in a recent (less than 7 days ago) round
- [ ] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

yell at me i don't give a fuck i'm gonna freewheel it on outta this issue

### PR DESCRIPTION
**Pull request by ultraplen**
fixes #12702 with the changes suggested by @Shadowmech88

Vocaroo previewaroo:
http://vocaroo.com/i/s1FdOAfVxbLJ

🆑

 * soundadd: Added go to mag drop sound to weapons with the AUTOMAGDROP flag

### CODE REVIEW

File: code/modules/projectiles/guns/projectile.dm [hunk 1]
```diff
@@ -225,7 +226,7 @@
 	..()
 	if(!chambered && stored_magazine && !stored_magazine.ammo_count() && gun_flags &AUTOMAGDROP) //auto_mag_drop decides whether or not the mag is dropped once it empties
 		RemoveMag(user)
-		playsound(user, 'sound/weapons/smg_empty_alarm.ogg', 40, 1)
+		playsound(user, mag_drop_sound, 40, 1)
```
Comment (ComicIronic) [inline code/modules/projectiles/guns/projectile.dm]: Add a check to see if `mag_drop_sound` is set - I might want to make a gun play no noise.

Comment (Shadowmech88) [reply code/modules/projectiles/guns/projectile.dm]: Does calling `playsound()` with a null sound runtime?

Comment (ComicIronic) [reply code/modules/projectiles/guns/projectile.dm]: Looking at the code, it will.

Comment (Shadowmech88) [reply code/modules/projectiles/guns/projectile.dm]: I just tested it locally and it didn't.
```
var/testing = null
playsound(src, testing, 100)
```
yielded no runtime errors.

Comment (ComicIronic) [reply code/modules/projectiles/guns/projectile.dm]: Weird, then the inbuilt Byond `sound()` proc accepts nulls and gives an empty sound in return, for some reason.

Comment (Probe1) [reply code/modules/projectiles/guns/projectile.dm]: Does this require changes?

Comment (ultraplen) [reply code/modules/projectiles/guns/projectile.dm]: @Probe1 I dont think so, if playsound doesnt runtime for null sounds its probably fine.

Comment (PJB3005) [reply code/modules/projectiles/guns/projectile.dm]: Doesn't null sound make it stop playing a sound on a channel?

Comment (Shadowmech88) [reply code/modules/projectiles/guns/projectile.dm]: Yes, but only if a channel is specified. I don't believe it does anything otherwise.

Comment (Probe1) [reply code/modules/projectiles/guns/projectile.dm]: I'm scared.

Comment (ultraplen) [reply code/modules/projectiles/guns/projectile.dm]: @Probe1 https://www.youtube.com/watch?v=HCzy5n-8_0U

Comment (Probe1) [reply code/modules/projectiles/guns/projectile.dm]: But whenever I used to play oggs if someone fired an energy weapon it'd shut off the song for everyone.

I hate sound. 

Test it and it'll be merged.

Comment (ultraplen) [reply code/modules/projectiles/guns/projectile.dm]: @Probe1 
Doing 
`				playsound(user, 'sound/weapons/Laser.ogg', 40, 1)
				playsound(user, null, 40, 1)`
did result in silence, I´ll have to add the check in I guess

Comment (Probe1) [reply code/modules/projectiles/guns/projectile.dm]: I hate that shit.  I feel you.

Comment (ultraplen) [reply code/modules/projectiles/guns/projectile.dm]: The deed is done

**Comment by Probe1** [comment]
@ComicIronic how come you never hang out in discord with us

**Comment by ComicIronic** [comment]
Discord is where fat babies go to die.

**Comment by Probe1** [comment]
Its also where we communicate and organize and i wanted to share the gift of laughter about merging a PR at the same time as you

**Pull request merged**